includes/multi-currency/FrontendCurrencies.php

```diff
@@ -29,22 +29,40 @@ class FrontendCurrencies {
 	 */
 	protected $localization_service;
 
+	/**
+	 * Multi-currency utils instance.
+	 *
+	 * @var Utils
+	 */
+	protected $utils;
+
 	/**
 	 * Multi-Currency currency formatting map.
 	 *
 	 * @var array
 	 */
 	protected $currency_format = [];
 
+
+	/**
+	 * Order currency code.
+	 *
+	 * @var string
+	 */
+	protected $order_currency;
+
 	/**
 	 * Constructor.
 	 *
 	 * @param MultiCurrency                    $multi_currency       The MultiCurrency instance.
 	 * @param WC_Payments_Localization_Service $localization_service The Localization Service instance.
+	 * @param Utils                            $utils                Utils instance.
 	 */
-	public function __construct( MultiCurrency $multi_currency, WC_Payments_Localization_Service $localization_service ) {
+	public function __construct( MultiCurrency $multi_currency, WC_Payments_Localization_Service $localization_service, Utils $utils ) {
 		$this->multi_currency       = $multi_currency;
 		$this->localization_service = $localization_service;
+		$this->utils                = $utils;
+		$this->init_order_currency();
```

Comment:
I know it's not optimal to init it on all pages, but it's errors with maximum nesting level 256 reached if `wc_get_order ` is called inside any `add_filter` callback.
I could add check `if ( $this->should_override_currency_code() )`, then it will init only in this case.

Would appreciate your thoughts how it can be further improved.

Reply:
@jessepearson helped me to nail this one ❤️